src/main/java/frc/robot/maps/subsystems/ElevatorMap.java

```diff
@@ -55,38 +58,49 @@ public double getValue(ElevatorPresets preset) {
     public final SmartMotorController leftMotor;
     public final SmartMotorController rightMotor;
     public final IEncoder encoder;
-    public final double conversionRate;
     public final ElevatorPresetValues elevatorPreset;
     public final ValueRange hardLimits;
     public final ValueRange softLimits;
+    public final ProfiledPIDController pid;
+    public final ElevatorFeedforward feedForward;
 
     public ElevatorMap() {
-        this(new SmartMotorController(), new SmartMotorController(), new MockEncoder(), 1, new ElevatorPresetValues(),
-                new ValueRange(0, 0), new ValueRange(0, 0));
+        this(new SmartMotorController(), new SmartMotorController(), new MockEncoder(), new ElevatorPresetValues(),
+                new ValueRange(0, 0), new ValueRange(0, 0), new ProfiledPIDController(0, 0, 0, new Constraints(0, 0)),
+                new ElevatorFeedforward(0, 0, 0));
     }
 
     public ElevatorMap(SmartMotorController leftMotor, SmartMotorController rightMotor, IEncoder encoder,
-            double conversionRate, ElevatorPresetValues elevatorPreset, ValueRange hardLimits, ValueRange softLimits) {
+            ElevatorPresetValues elevatorPreset, ValueRange hardLimits, ValueRange softLimits,
+            ProfiledPIDController pid, ElevatorFeedforward feedForward) {
         this.leftMotor = leftMotor;
         this.rightMotor = rightMotor;
         this.encoder = encoder;
-        this.conversionRate = conversionRate;
         this.elevatorPreset = elevatorPreset;
         this.hardLimits = hardLimits;
         this.softLimits = softLimits;
+        this.pid = pid;
+        this.feedForward = feedForward;
     }
 
     @Override
     public void updateData(Data data) {
         data.leftMotor.updateData(leftMotor);
         data.rightMotor.updateData(rightMotor);
-        data.heightAbsInches = encoder.getAbsolutePosition() * conversionRate; // need to do math to figure out the
-                                                                               // right value
+        data.heightAbsInches = encoder.getAbsolutePosition(); // need to do math to figure out the
+                                                              // right value
+        data.liftingHeightVelocity = encoder.getRate();
     }
 
     public static class Data extends DataWrapper {
         public MotorControllerData leftMotor = new MotorControllerData();
         public MotorControllerData rightMotor = new MotorControllerData();
         public double heightAbsInches;
+        public double liftingHeightVelocity;
+
+        public void setSetpoint(double setpoint) {
+            leftMotor.setpoint = setpoint;
+            rightMotor.setpoint = setpoint;
+        }
```

Comment:
I don't think we need this. The elevator subsystem can just do `getData().motor.setpoint = X`